utils/cmut__utils__compile_resource.cmake

```diff
@@ -0,0 +1,139 @@
+    else ()
+        set(${FILE_NAME} "${OUTPUT_FOLDER}/GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" PARENT_SCOPE)
+    endif ()
+endfunction()
+
+function(cmut__utils__get_source_name FILE_NAME INPUT_FILE OUTPUT_FOLDER)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    # create Name ouf output files .h & .cpp
+    if ("${OUTPUT_FOLDER}" STREQUAL "")
+        set(${FILE_NAME} "GENREATED__${SOURCE_FILE_NAME_VARIABLE}.cpp" PARENT_SCOPE)
+    else ()
+        set(${FILE_NAME} "${OUTPUT_FOLDER}/GENREATED__${SOURCE_FILE_NAME_VARIABLE}.cpp" PARENT_SCOPE)
+    endif ()
+endfunction()
+
+
+function(cmut__utils__get_header_lock LOCK_VALUE INPUT_FILE)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    # create The header multiple include lock
+    string(TOUPPER "GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" HEADER_LOCK)
+    string(REGEX REPLACE "[^A-Z0-9]" "_" HEADER_LOCK "${HEADER_LOCK}")
+    # create Name ouf output files .h & .cpp
+    set(${LOCK_VALUE} "${HEADER_LOCK}" PARENT_SCOPE)
+endfunction()
+
+
+function(cmut__utils__create_header_from_file INPUT_FILE OUTPUT_FOLDER)
+    # Check if the input file exist:
+    if ( NOT DEFINED INPUT_FILE )
+        message( FATAL_ERROR "cmut__utils__create_header_from_file: File does not exist: '${INPUT_FILE}'" )
+    endif()
+    cmut_info(" cmut__utils__create_header_from_file: File: '${INPUT_FILE}'" )
+    # Read the data form the input file
+    file(READ "${INPUT_FILE}" contents HEX)
+    get_filename_component(SOURCE_FILE_NAME ${INPUT_FILE} NAME)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    cmut__utils__get_header_name(FILE_HEADER ${INPUT_FILE} ${OUTPUT_FOLDER})
+    cmut__utils__get_source_name(FILE_SOURCE ${INPUT_FILE} ${OUTPUT_FOLDER})
+    cmut__utils__get_header_lock(HEADER_LOCK ${INPUT_FILE})
+    # Generate all the HEADER file:
+    cmut_info(" Generate file: '${FILE_HEADER}'" )
+    file(WRITE "${FILE_HEADER}" "/** @file\n")
+    file(APPEND "${FILE_HEADER}" " * @brief Include header in a C++ header\n")
+    file(APPEND "${FILE_HEADER}" " * @copyright UNOWHY 2017\n")
+    file(APPEND "${FILE_HEADER}" " * @licence PROPRIETARY\n")
+    file(APPEND "${FILE_HEADER}" " * @note Auto-generated file with cmake\n")
+    file(APPEND "${FILE_HEADER}" " */\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "#ifndef ${HEADER_LOCK}\n")
+    file(APPEND "${FILE_HEADER}" "#define ${HEADER_LOCK}\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "#include <string>\n")
+    file(APPEND "${FILE_HEADER}" "#include <vector>\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "/**\n")
+    file(APPEND "${FILE_HEADER}" " * @brief Name space of all auto-generateFile\n")
+    file(APPEND "${FILE_HEADER}" " */\n")
+    file(APPEND "${FILE_HEADER}" "namespace generateFile {\n")
```

Comment:
add optional NAMESPACE parameter to define custom namespace

Reply:
TODO ...